### ISSUE TITLE
Intel Flash Descriptor v2 Regions

### ISSUE THREAD

**Issue by platomav** [LongSoft/UEFITool#89]
For Intel FD v2, these are the names of all regions in order:

Flash Descriptor
BIOS
Management Engine
Gigabit Ethernet
Platform Data
Device Expansion 1
Secondary BIOS
CPU Microcode
Embedded Controller
Device Expansion 2
Innovation Engine
10 Gigabit Ethernet 1
10 Gigabit Ethernet 2
Reserved 1
Reserved 2
Platform Trust Technology

Can you add parsing for them with the proper names as shown above?

**Comment by skochinsky** [comment]
how do you detect descriptor v2?

**Comment by platomav** [comment]
There is no actual indication from Intel but FLCOMP > Read Clock Frequency has proven to work just fine so far. It is always 20MHz for FD v1 and 17MHz for FD v2.

**Comment by NikolajSchlej** [comment]
Any sample files with those?

**Comment by platomav** [comment]
Any Skylake+ SPI uses FD v2. All Apollo Lake SPI use Device Expansion 1 (called Reserved 1 at UT). I recently found a server SPI image from HP which had firmware at the Innovation Engine. I don't have the link anymore but I'll look. As for the order/naming, it is correct based on spsFIT v4. One strange thing is that the last partition is called Reserved 3 at one place and PTT at another. I went with PTT.
